holmes/core/toolset_manager.py

```diff
@@ -0,0 +1,394 @@
+            toolset_status = [
+                json.loads(
+                    toolset.model_dump_json(
+                        include={"name", "status", "enabled", "type", "path", "error"}
+                    )
+                )
+                for toolset in all_toolsets
+            ]
+            json.dump(toolset_status, f, indent=2)
+        logging.info(f"Toolset statuses are cached to {self.toolset_status_location}")
+
+    def load_toolset_with_status(
+        self, dal: Optional[SupabaseDal] = None, refresh_status: bool = False
+    ) -> List[Toolset]:
+        """
+        Load the toolset status from the cache file.
+        If the file does not exist, return an empty list.
+        """
+
+        if not os.path.exists(self.toolset_status_location) or refresh_status:
+            logging.info("refreshing toolset status")
+            self.refresh_toolset_status(dal)
+
+        cached_toolsets: List[dict[str, Any]] = []
+        with open(self.toolset_status_location, "r") as f:
+            cached_toolsets = json.load(f)
+
+        # load status from cached file and update the toolset details
+        toolsets_status_by_name: dict[str, dict[str, Any]] = {
+            cached_toolset["name"]: cached_toolset for cached_toolset in cached_toolsets
+        }
+        all_toolsets_with_status = self._list_all_toolsets(
+            dal=dal, check_prerequisites=False
+        )
+
+        for toolset in all_toolsets_with_status:
+            if toolset.name in toolsets_status_by_name:
+                # Update the status and error from the cached status
+                cached_status = toolsets_status_by_name[toolset.name]
+                toolset.status = ToolsetStatusEnum(cached_status["status"])
+                toolset.error = cached_status.get("error", None)
+                toolset.enabled = cached_status.get("enabled", True)
+                toolset.type = ToolsetType(
+                    cached_status.get("type", ToolsetType.BUILTIN)
+                )
+                toolset.path = cached_status.get("path", None)
+
+        # CLI custom toolsets status are not cached, and their prerequisites are always checked whenever the CLI runs.
+        custom_toolsets_from_cli = self._load_toolsets_from_paths(
+            self.custom_toolsets_from_cli,
+            list(toolsets_status_by_name.keys()),
+            check_conflict_default=True,
+        )
+        # custom toolsets from cli as experimental toolset should not override custom toolsets from config
+        for custom_toolset_from_cli in custom_toolsets_from_cli:
+            if custom_toolset_from_cli in toolsets_status_by_name.keys():
+                raise Exception(
+                    f"Toolset {custom_toolset_from_cli.name} from cli is already defined in existing toolset"
+                )
+        all_toolsets_with_status.extend(custom_toolsets_from_cli)
+
+        return all_toolsets_with_status
+
+    def list_enabled_console_toolsets(
```

Comment:
The current implementation cause it to disable all toolset by default.
This is different behiour then our initial intantion. We would like that every new user will have toolsets that are enabled by default to start and get the hand of holmes before messing with toolsets config. This is quite a change and i think that we need to be aligned on that.
@arikalon1 WDYT?

Reply:
Thanks for sharing the intention of the design.

My intention is to disable all tools by default, and customer has to enable the tools explicitly. This allows us to reduce the chance the loading unprepared toolset.